src/pool_worker_master.erl

```diff
@@ -0,0 +1,49 @@
+-module(pool_worker_master).
+% using list_to_integer as command line env variable declaration looks to set string
+-define(WORKER_NUMBER, list_to_integer(os:getenv("WORKER_NUMBER", "4"))).
```

Comment:
This is an interesting line of code.

I think `WORKER_NUMBER` is not a constant, evaluated once, but a macro. Which means, every time we use `?WORKER_NUMBER`, the expression is inlined. So `os:getenv` is called as many time as `?WORKER_NUMBER` is evaluated.

Reply:
My tests confirm that!

If you use `erlc` with `-P` argument, you will see the files with macros expanded.

```sh
erlc -P src/*.erl
```

![CleanShot 2022-06-17 at 19 14 34@2x](https://user-images.githubusercontent.com/29370468/174346677-3016dd68-0ede-471f-b964-0615f4ea09fe.png)

Reply:
Interesting :open_mouth: